src/bin/pg_autoctl/file_utils.c

```diff
@@ -768,18 +699,15 @@ set_program_absolute_path(char *program, int size)
 		 * Now either return pg_autoctl_argv0 when that's an absolute filename,
 		 * or search for it in the PATH otherwise.
 		 */
-		char **pathEntries = NULL;
-		int n;
+		SearchPath paths = { 0 };
 
 		if (pg_autoctl_argv0[0] == '/')
 		{
 			strlcpy(program, pg_autoctl_argv0, size);
 			return true;
 		}
 
-		n = search_path(pg_autoctl_argv0, &pathEntries);
-
-		if (n < 1)
+		if (!search_path(pg_autoctl_argv0, &paths))
```

Comment:
should check `paths.found` too.

Reply:
You did that, thanks!

Reply:
Not in this one, I did it in `search_path_first` only to fix the monitor creation for me. And then I noticed the same issue here during review.

Reply:
Ah. Indeed. Sorry about that, done now.